Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionFragment.kt

```diff
@@ -1,24 +1,120 @@
+            certificatesAdapter.update(certificatesList)
+
+            binding.apply {
+                if (certificatesAdapter.itemCount == 0) {
+                    selectionTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_no_certificates_found_title)
+                    noCertificatesFound.visibility = View.VISIBLE
+                    allowedCertificates.visibility = View.INVISIBLE
+                    birthDate.visibility = View.INVISIBLE
+                    standardizedName.visibility = View.INVISIBLE
+                    requiredCertificatesTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_requested_certificates_text)
+                    faqLink.visibility = View.VISIBLE
+                    val link = when (Locale.getDefault().language) {
+                        Locale.GERMAN.language -> R.string.dcc_ticketing_faq_link_german
+                        else -> R.string.dcc_ticketing_faq_link_english
+                    }
+                    faqLink.setTextWithUrl(
+                        R.string.dcc_ticketing_certificate_selection_more_information_text,
+                        R.string.dcc_ticketing_faq_link_container,
+                        link
+                    )
+                } else {
+                    selectionTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_provider_requirements_title)
+                    noCertificatesFound.visibility = View.GONE
+                    allowedCertificates.text = args.transactionContext.accessTokenPayload?.vc?.type?.let {
+                        getRequestedCertificateTypes(
+                            requestedCertificatesList = it,
+                            context = requireContext(),
+                            separator = ","
+                        )
+                    } ?: ""
+                    birthDate.text = getString(R.string.dcc_ticketing_certificate_birthday)
+                        .format(args.transactionContext.accessTokenPayload?.vc?.dob)
+                    standardizedName.text = getFullName(
+                        args.transactionContext.accessTokenPayload?.vc?.fnt,
+                        args.transactionContext.accessTokenPayload?.vc?.gnt
+                    )
+                    requiredCertificatesTitle.text =
+                        getString(R.string.dcc_ticketing_certificates_meeting_requirements_title)
+                }
+            }
+        }
+    }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        // TODO Pass args.transactionContext.accessTokenPayload to the view model
-        return inflater.inflate(R.layout.fragment_dcc_ticketing_certificate_selection, container, false)
+    private fun returnToScreenWhereScannerWasOpened() {
```

Comment:
Do we need to do this ? 

I think when you navigate from `consentI` to selection screen and pop to `consentI`

1- `ConsentI ------- popuo to consentI -> selection screen` 
2- Close in selection screen just use `popBackStack ` it will work and land where use started